user/uploading-artifacts.md

```diff
@@ -5,7 +5,7 @@ layout: en
 ---
 
 Travis CI can automatically upload your build artifacts to Amazon S3, after the
-[`after success`](/user/job-lifecycle/) stage of the
+[`after success` or `after_failure`](/user/job-lifecycle/) phases of the
```

Comment:
What about keeping it a simple:

```suggestion
end of the job.
```

instead?

Reply:
We're not entirely consistent, but we include the deploy steps in the [job lifecycle](https://docs.travis-ci.com/user/job-lifecycle/) which is probably why I was specific there.